### PR TITLE
feat/Header,Sidebar

### PR DESCRIPTION
**Pull request by DucBinh1207**
- Use Coockie.js
- Feat/Header ( Responsive)
- Feat/Sidebar ( Responsive)
- Trigger Sidebar by Zustand

### CODE REVIEW

File: src/components/common/HeaderNotice/page.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          </a>
+        </li>
+        <li className="top-header_hover relative flex flex-nowrap items-center">
+          <FontAwesomeIcon icon={faBusinessTime} className="mr-2" />
+          Mon-Fri: 8:00 AM - 20:00 PM
+        </li>
+      </ul>
+    </div>
+  );
+}
```
Comment (ghost) [inline src/components/common/HeaderNotice/page.tsx]: tạo component

File: src/app/globals.css [hunk 2]
```diff
@@ -16,15 +40,14 @@
...
-    rgb(var(--background-start-rgb));
-} */
+body {
+  font-family: var(--font-outfit);
+  box-sizing: border-box;
+}
+
+html {
+  font-size: 125%;
+}
```
Comment (ghost) [inline src/app/globals.css]: ??

File: src/app/loading.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function Loading() {
+  return <div>Loading . . . </div>;
+}
```
Comment (ghost) [inline src/app/loading.tsx]: add loading

File: src/app/login/page.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import Button from "../../_components/common/Button/page";
-import Input from "../../_components/common/Input/page";
+import Button from "../../components/common/Button/page";
+import Input from "../../components/common/Input/page";
```
Comment (ghost) [inline src/app/login/page.tsx]: alias

File: src/components/common/Header/page.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  const { openSidebar } = triggerSidebar((state) => state);
+
+  return (
+    <div className="relative">
+      <div className="sticky top-0 flex min-h-[90px] w-full items-center justify-between bg-header_bg px-[50px] after:absolute after:top-full after:block after:h-[3px] after:w-full after:translate-x-[-50px] after:bg-header_img after:bg-repeat-x after:content-[''] small-screen:min-h-[60px] small-screen:justify-normal small-screen:p-0 small-screen:after:translate-x-0">
+        <button
+          className="hidden h-[60px] w-[86px] flex-col gap-[5px] text-white small-screen:block"
+          onClick={() => {
+            openSidebar();
+          }}
```
Comment (ghost) [inline src/components/common/Header/page.tsx]:
```suggestion
          onClick={openSidebar}
```

File: src/components/common/Header/page.tsx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+        <ul className="text-navbar-size flex text-header_text small-screen:hidden small-screen:opacity-0">
+          <li className="mx-[27.5px] my-[15px]">
+            <Link
+              href="/"
+              className={`group relative flex hover:!text-header_text_third ${
+                pathName === "/"
+                  ? "after:absolute after:bottom-[-10px] after:left-0 after:h-[1px] after:w-full after:bg-header_text after:content-['']"
+                  : ""
+              }`}
```
Comment (ghost) [inline src/components/common/Header/page.tsx]: tạo component

File: src/components/common/HeaderNotice/page.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faLocationDot,
+  faBusinessTime,
+  faPhone,
+  faEnvelope,
+} from "@fortawesome/free-solid-svg-icons";
+export default function HeaderNotice() {
```
Comment (ghost) [inline src/components/common/HeaderNotice/page.tsx]: line

File: src/components/common/Sidebar/page.tsx [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+import Pet from "/src/assets/icons/dog_icon.svg";
+import Food from "/src/assets/icons/food_icon.svg";
+import Supplies from "/src/assets/icons/supplies_icon.svg";
+import Guide from "/src/assets/icons/guide_icon.svg";
+import { usePathname } from "next/navigation";
+import triggerSidebar from "@/store/trigger-sidebar";
+
+export default function Sidebar() {
+  const pathName = usePathname();
+  const { sidebar, closeSidebar } = triggerSidebar((state) => ({
```
Comment (ghost) [inline src/components/common/Sidebar/page.tsx]: useSidebar

File: src/components/common/Sidebar/page.tsx [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+      className={`${sidebar === true ? "block opacity-100" : "hidden opacity-0"} fixed top-0 z-[1000] h-full w-full min-w-[320px] max-w-[415px] bg-header_bg_second text-header_text`}
+    >
+      <div className="absolute left-0 top-0 flex h-[60px] w-full items-center justify-end px-[25px]">
+        <button>
+          <FontAwesomeIcon
+            icon={faXmark}
+            size="2x"
+            onClick={() => {
+              closeSidebar();
+            }}
```
Comment (ghost) [inline src/components/common/Sidebar/page.tsx]:
```suggestion
            onClick={closeSidebar}
```

File: src/store/trigger-sidebar.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { create } from "zustand";
+
+type SidebarStore = {
+  sidebar: boolean;
+  openSidebar: () => void;
+  closeSidebar: () => void;
+};
+
```
Comment (ghost) [inline src/store/trigger-sidebar.ts]:
```suggestion
type SidebarStore = {
  isOpen: boolean;
  openSidebar: () => void;
  closeSidebar: () => void;
};

```

File: src/app/_components/Header/_components/Nav-item/page.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
+import classNameComp from "@/components/common/Class";
+import { ShopIcon } from "@/components/common/Icons";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+
+export default function NavItem() {
+  const pathName = usePathname();
+
+  return (
+    <>
```
Comment (dungnq-1120) [inline src/app/_components/Header/_components/Nav-item/page.tsx:10]:
```suggestion
    <>
```

Comment (ghost) [inline src/app/_components/Header/_components/Nav-item/page.tsx]:
```suggestion

```

File: src/app/_components/Header/_components/Nav-item/page.tsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+            alt="shop page"
+            loading="lazy"
+            className={classNameComp(
+              "mr-[16px] brightness-100 group-hover:brightness-[1.5]",
+              {
+                "brightness-[1.5]": pathName === "/",
+              },
+            )}
+          /> */}
+          Svg Shop
```
Comment (ghost) [inline src/app/_components/Header/_components/Nav-item/page.tsx]: xoá

File: src/app/_components/Header/_components/Nav-item/page.tsx [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+          <span
+            className={`absolute bottom-[-10px] left-[50%] h-[1px] w-0 origin-center translate-x-[-50%] bg-header_text transition-all duration-300 group-hover:w-full`}
+          ></span>
```
Comment (ghost) [inline src/app/_components/Header/_components/Nav-item/page.tsx]: template string/

File: src/app/_components/Header/page.tsx [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+    if (window.scrollY >= 50) setHeaderSticky(true);
+    else setHeaderSticky(false);
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", scrollHandle);
+    return () => {
+      window.addEventListener("scroll", scrollHandle);
+    };
+  }, []);
```
Comment (ghost) [inline src/app/_components/Header/page.tsx]: tạo ra 1 hook riêng

Comment (ghost) [reply src/app/_components/Header/page.tsx]: useClickOutside

File: src/app/_components/Header/page.tsx [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+
+  return (
+    <div className="relative">
+      <div
+        className={`${headerSticky === true ? "fixed" : "large-screen:relative"} top-0 z-[100] flex min-h-[90px] w-full items-center justify-between bg-header_bg px-[50px] after:absolute after:top-full after:block after:h-[3px] after:w-full after:translate-x-[-50px] after:bg-header_img after:bg-repeat-x after:content-[''] small-screen:min-h-[60px] small-screen:justify-normal small-screen:p-0 small-screen:after:translate-x-0`}
```
Comment (ghost) [inline src/app/_components/Header/page.tsx]: cn

File: src/app/_components/Sidebar/page.tsx [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+
+    if (isSideBarOpen) {
+      document.addEventListener("mousedown", handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [isSideBarOpen, closeSidebar]);
+
```
Comment (ghost) [inline src/app/_components/Sidebar/page.tsx]: clickoutside

File: src/app/fontawesome.tsx [hunk 17]
```diff
@@ -0,0 +1,7 @@
+import { config } from "@fortawesome/fontawesome-svg-core";
+import "@fortawesome/fontawesome-svg-core/styles.css";
+config.autoAddCss = false;
+
+export default function FontAwesomeConfig() {
+  return null;
+}
```
Comment (ghost) [inline src/app/fontawesome.tsx]: xoá

File: src/app/globals.css [hunk 18]
```diff
@@ -2,6 +2,57 @@
...
+  @apply py-[15px] text-[17px] font-semibold uppercase leading-[1.23];
+}
+
+.icon_sidebar_animate {
+  @apply transform transition-transform;
+  transition:
+    transform 0.4s ease-in-out,
+    opacity 0.4s ease-in-out;
+}
+
```
Comment (ghost) [inline src/app/globals.css]: bỏ component

File: src/app/layout.tsx [hunk 19]
```diff
@@ -20,7 +46,19 @@ export default function RootLayout({
 }>) {
   return (
     <html lang="en">
-      <body className={outfit_init.variable}>{children}</body>
+      <head>
+        <FontAwesomeConfig />
```
Comment (ghost) [inline src/app/layout.tsx]: xoá

File: src/app/loading.tsx [hunk 20]
```diff
@@ -0,0 +1,14 @@
+"use client";
+
+import Lottie from "lottie-react";
+import LoadingCircle from "../assets/animations/loading-circle.json";
+
+export default function Loading() {
+  return (
+    <>
```
Comment (ghost) [inline src/app/loading.tsx]: xoá, dư

File: src/app/page.tsx [hunk 21]
```diff
@@ -1,101 +1,9 @@
...
-            src="https://nextjs.org/icons/globe.svg"
-            alt="Globe icon"
-            width={16}
-            height={16}
-          />
-          Go to nextjs.org →
-        </a>
-      </footer>
+    <div>
+      <NotFound />
```
Comment (ghost) [inline src/app/page.tsx]:
```suggestion
      <NotFound />
```

File: src/components/common/Button/page.tsx [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+    variant: {
+      primary:
+        "bg-primary text-white border-primary hover:bg-white hover:text-primary",
+
+      secondary:
+        "bg-white text-primary border-primary hover:bg-primary hover:text-white",
+
+      third:
+        "bg-white text-primary border-white hover:bg-primary hover:border-primary hover:text-white",
+    },
```
Comment (ghost) [inline src/components/common/Button/page.tsx]: primary - Primary
secondary - Secondary
success - Success
warning - Warning
danger - Danger
info - Info
light - Light
dark - Dark

File: src/components/common/Button/page.tsx [hunk 23]
```diff
@@ -0,0 +1,80 @@
...
+  children,
+  ...rest
+}: ButtonProps) {
+  let btnClassName =
+    "inline-block outline-none border-solid border-[2px] cursor-pointer rounded-[25px] text-center uppercase ";
+
+  const classVariants = useMemo(() => {
+    return buttonVariants({
+      size: size ?? undefined,
+      variant: variant ?? undefined,
```
Comment (ghost) [inline src/components/common/Button/page.tsx]:
```suggestion
      size: size,
      variant: variant
```

File: src/components/common/Button/page.tsx [hunk 24]
```diff
@@ -0,0 +1,80 @@
...
+    });
+  }, [size, variant]);
+
+  (size || variant) && (btnClassName += ` ${classVariants}`);
+  className && (btnClassName += ` ${className}`);
+
```
Comment (ghost) [inline src/components/common/Button/page.tsx]: const startIcon = startIconProp && <span className="inline-flex items-center">{startIcon}<span>

File: src/app/not-found.tsx [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+            <Input placeholder="Search..." className="px-[12px] py-[9px]" />
+          </div>
+          <ToolTip
+            value="Search . . . "
+            button={
+              <Button
+                className="btn_icon_hover_animate h group peer relative flex-1 border-primary bg-white px-[15px] py-[0px] text-[16px] leading-[0px] tracking-[0.025em] hover:bg-primary"
+                startIcon={
+                  <div className="inline bg-primary group-hover:bg-white">
+                    <SearchIcon size="small" />
```
Comment (ghost) [inline src/app/not-found.tsx]:
```suggestion
                    <SearchIcon size="small" className/>
```

File: src/components/common/Icons/index.tsx [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+export function ShopIcon(props: SvgProps) {
+  return (
+    <Svg stroke="currentColor" viewBox="0 0 24 24" {...props}>
+      <g id="SVGRepo_bgCarrier" stroke-width="0" />
+
+      <g
+        id="SVGRepo_tracerCarrier"
+        stroke-linecap="round"
+        stroke-linejoin="round"
+      />
```
Comment (ghost) [inline src/components/common/Icons/index.tsx]: export {
  iconA, iconB }

File: src/components/common/Icons/index.tsx [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+          fill="currentColor"
+        />
+        <path
+          fill-rule="evenodd"
+          clip-rule="evenodd"
+          d="M20 21.25H22C22.4142 21.25 22.75 21.5858 22.75 22C22.75 22.4142 22.4142 22.75 22 22.75H2C1.58579 22.75 1.25 22.4142 1.25 22C1.25 21.5858 1.58579 21.25 2 21.25H4L4 12.5C4.74363 12.5 5.43309 12.2681 6 11.8727C6.56692 12.2681 7.25638 12.5 8 12.5C8.74363 12.5 9.43309 12.2681 10 11.8727C10.5669 12.2681 11.2564 12.5 12 12.5C12.7436 12.5 13.4331 12.2681 14 11.8727C14.5669 12.2681 15.2564 12.5 16 12.5C16.7436 12.5 17.4331 12.2681 18 11.8727C18.5669 12.2681 19.2564 12.5 20 12.5L20 21.25ZM9.5 21.25H14.5V18.5C14.5 17.5654 14.5 17.0981 14.299 16.75C14.1674 16.522 13.978 16.3326 13.75 16.2009C13.4019 16 12.9346 16 12 16C11.0654 16 10.5981 16 10.25 16.2009C10.022 16.3326 9.83261 16.522 9.70096 16.75C9.5 17.0981 9.5 17.5654 9.5 18.5V21.25Z"
+          fill="currentColor"
+        />{" "}
```
Comment (ghost) [inline src/components/common/Icons/index.tsx]:
```suggestion
        />
```

File: src/components/common/Icons/index.tsx [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+    <Svg fill="none" viewBox="0 0 512 512" {...props}>
+      <path
+        fill="currentColor"
+        d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
+      />
+    </Svg>
+  );
+}
+
+export function ShopIcon(props: SvgProps) {
```
Comment (ghost) [inline src/components/common/Icons/index.tsx]: shop-icon.tsx

File: src/components/common/Icons/index.tsx [hunk 29]
```diff
@@ -0,0 +1,39 @@
...
+          fill="currentColor"
+        />
+        <path
+          fill-rule="evenodd"
+          clip-rule="evenodd"
+          d="M20 21.25H22C22.4142 21.25 22.75 21.5858 22.75 22C22.75 22.4142 22.4142 22.75 22 22.75H2C1.58579 22.75 1.25 22.4142 1.25 22C1.25 21.5858 1.58579 21.25 2 21.25H4L4 12.5C4.74363 12.5 5.43309 12.2681 6 11.8727C6.56692 12.2681 7.25638 12.5 8 12.5C8.74363 12.5 9.43309 12.2681 10 11.8727C10.5669 12.2681 11.2564 12.5 12 12.5C12.7436 12.5 13.4331 12.2681 14 11.8727C14.5669 12.2681 15.2564 12.5 16 12.5C16.7436 12.5 17.4331 12.2681 18 11.8727C18.5669 12.2681 19.2564 12.5 20 12.5L20 21.25ZM9.5 21.25H14.5V18.5C14.5 17.5654 14.5 17.0981 14.299 16.75C14.1674 16.522 13.978 16.3326 13.75 16.2009C13.4019 16 12.9346 16 12 16C11.0654 16 10.5981 16 10.25 16.2009C10.022 16.3326 9.83261 16.522 9.70096 16.75C9.5 17.0981 9.5 17.5654 9.5 18.5V21.25Z"
```
Comment (ghost) [inline src/components/common/Icons/index.tsx]: https://www.svgminify.com/

File: src/components/common/Svg/index.tsx [hunk 30]
```diff
@@ -0,0 +1,32 @@
...
+  large: 32,
+};
+
+type SvgProps = ComponentProps<"svg"> & {
+  size: Size;
+};
+
+function Svg({ size: sizeProp, className, ...rest }: SvgProps) {
+  const size = typeof sizeProp === "string" ? sizesMap[sizeProp] : sizeProp;
+  return (
```
Comment (ghost) [inline src/components/common/Svg/index.tsx]: line

File: src/components/common/Tooltip/_components/page.tsx [hunk 31]
```diff
@@ -0,0 +1,15 @@
+type TooltipProps = {
+  value?: string;
+};
+
+export default function TooltipNode({ value }: TooltipProps) {
+  return (
+    <>
```
Comment (ghost) [inline src/components/common/Tooltip/_components/page.tsx]: xoá

File: src/components/common/Tooltip/_components/page.tsx [hunk 32]
```diff
@@ -0,0 +1,15 @@
+type TooltipProps = {
+  value?: string;
+};
+
+export default function TooltipNode({ value }: TooltipProps) {
```
Comment (ghost) [inline src/components/common/Tooltip/_components/page.tsx]:
```suggestion
export default function TooltipNode({ value, refElm }: TooltipProps) {
```

File: src/store/useSidebar.ts [hunk 33]
```diff
@@ -0,0 +1,15 @@
...
+  isSideBarOpen: boolean;
+  openSidebar: () => void;
+  closeSidebar: () => void;
+};
+
+const triggerSidebar = create<SidebarStore>((set) => ({
+  isSideBarOpen: false,
+  openSidebar: () => set({ isSideBarOpen: true }),
+  closeSidebar: () => set({ isSideBarOpen: false }),
+}));
```
Comment (ghost) [inline src/store/useSidebar.ts]: isOpen

File: src/components/common/Class/index.tsx [hunk 34]
```diff
@@ -0,0 +1,6 @@
+import { ClassValue } from "class-variance-authority/types";
+import clsx from "clsx";
+
+export default function classNameComp(...inputClassName: ClassValue[]) {
+  return clsx(inputClassName);
+}
```
Comment (ghost) [inline src/components/common/Class/index.tsx]: move utils/style/cn.ts

File: src/app/_components/Header/_components/Action-item/page.tsx [hunk 35]
```diff
@@ -0,0 +1,42 @@
...
+
+const IconsMap = {
+  search: SearchIcon,
+  user: UserIcon,
+  cart: CartIcon,
+};
+
+export default function ActionItem({ type, iconName, href }: ActionProps) {
+  const Icon = IconsMap[iconName];
+  href = href !== undefined ? href : "";
```
Comment (ghost) [inline src/app/_components/Header/_components/Action-item/page.tsx]:
```suggestion
export default function ActionItem({ type, iconName, href: hrefProp }: ActionProps) {
  const Icon = IconsMap[iconName];
 const href = hrefProp ?? ""
```

File: src/app/_components/Header/_components/Nav-item/page.tsx [hunk 36]
```diff
@@ -0,0 +1,63 @@
...
+          size={26.4}
+        />
+
+        {name}
+
+        <span
+          className={cn(
+            "absolute bottom-[-10px] left-[50%] h-[1px] w-0 origin-center translate-x-[-50%] bg-header_text transition-all duration-300 group-hover:w-full",
+          )}
+        ></span>
```
Comment (ghost) [inline src/app/_components/Header/_components/Nav-item/page.tsx]:
```suggestion
        <span
          className={absolute bottom-[-10px] left-[50%] h-[1px] w-0 origin-center translate-x-[-50%] bg-header_text transition-all duration-300 group-hover:w-full}
       />
```

File: src/app/_components/Header/page.tsx [hunk 37]
```diff
@@ -0,0 +1,73 @@
...
+  }, []);
+
+  return (
+    <div className="relative">
+      <div
+        className={cn(
+          "top-0 z-[100] flex min-h-[90px] w-full items-center justify-between bg-header_bg px-[50px] after:absolute after:top-full after:block after:h-[3px] after:w-full after:translate-x-[-50px] after:bg-header_img after:bg-repeat-x after:content-[''] small-screen:min-h-[60px] small-screen:justify-normal small-screen:p-0 small-screen:after:translate-x-0",
+          {
+            ["fixed"]: headerSticky === true,
+            ["large-screen:relative"]: headerSticky === false,
```
Comment (ghost) [inline src/app/_components/Header/page.tsx]:
```suggestion
            "fixed": headerSticky === true,
            "large-screen:relative": headerSticky === false,
```

File: .vscode/settings.json [hunk 38]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (ghost) [inline .vscode/settings.json]: xoa

File: src/app/_components/footer/_components/benefit-item/index.tsx [hunk 39]
```diff
@@ -0,0 +1,25 @@
...
+type props = {
+  value: string;
+  icon: "car" | "question" | "money";
+};
+
+const IconsMap = {
+  car: CarIcon,
+  question: QuestionMarkIcon,
+  money: MoneyIcon,
+};
```
Comment (ghost) [inline src/app/_components/footer/_components/benefit-item/index.tsx]:
```suggestion
const IconsMap = {
  CAR: CarIcon,
  question: QuestionMarkIcon,
  money: MoneyIcon,
};
```

File: src/app/_components/footer/_components/contact-item/index.tsx [hunk 40]
```diff
@@ -0,0 +1,39 @@
...
+  href: string;
+  icon: "mail" | "phone" | "locate" | "businessTime";
+};
+
+const IconsMap = {
+  mail: MailIcon,
+  phone: PhoneIcon,
+  locate: LocateIcon,
+  businessTime: BusinessTimeIcon,
+};
```
Comment (ghost) [inline src/app/_components/footer/_components/contact-item/index.tsx]: xx

File: src/app/_components/footer/_components/legal-item/index.tsx [hunk 41]
```diff
@@ -0,0 +1,14 @@
...
+
+export default function LegalItem({ value, href }: props) {
+  return (
+    <span className="smallest-screen:mr-[15px mb-[12px] mr-[10px] smallest-screen:mb-[18px]">
+      <a href={href} className="hover_animate hover:text-secondary">
+        {value}
+      </a>
+    </span>
+  );
+}
```
Comment (ghost) [inline src/app/_components/footer/_components/legal-item/index.tsx]: ko cần tách

File: src/app/_components/footer/_components/overview-item/index.tsx [hunk 42]
```diff
@@ -0,0 +1,23 @@
...
+  icon: "heart" | "car";
+};
+const IconsMap = {
+  heart: HeartWithDogIcon,
+  car: CarIcon,
+};
+
+export default function OverviewItem({ value, icon }: props) {
+  const Icon = IconsMap[icon];
+  return (
```
Comment (ghost) [inline src/app/_components/footer/_components/overview-item/index.tsx]: line

File: src/app/_components/header-notice/_components/notice-item/index.tsx [hunk 43]
```diff
@@ -0,0 +1,40 @@
...
+export default function NoticeItem({ name, href, iconName }: noticeItemProps) {
+  const Icon = IconsMap[iconName];
+
+  return (
+    <li className="relative flex flex-nowrap items-center justify-center border-b border-solid border-transparent hover:border-header_text">
+      <a
+        href={href}
+        className={cn(
+          "inline-flex text-[13px] font-semibold leading-[1.23] tracking-[0.015em] text-header_text",
+          {
```
Comment (ghost) [inline src/app/_components/header-notice/_components/notice-item/index.tsx]: thêm cmt

File: src/app/_components/header-notice/_components/notice-item/index.tsx [hunk 44]
```diff
@@ -0,0 +1,40 @@
...
+            "pointer-events-none": iconName === "businessTime",
```
Comment (ghost) [inline src/app/_components/header-notice/_components/notice-item/index.tsx]: if text-footer_icon

return span

or 
return a

File: src/components/common/Button/index.tsx [hunk 45]
```diff
@@ -0,0 +1,94 @@
...
+
+    const classVariants = useMemo(() => {
+      return buttonVariants({
+        size,
+        variant,
+      });
+    }, [size, variant]);
+
+    (size || variant) && (btnClassName += ` ${classVariants}`);
+    className && (btnClassName += ` ${className}`);
```
Comment (ghost) [inline src/components/common/Button/index.tsx]: sửa lại cho dễ hiểu

File: src/components/common/Button/index.tsx [hunk 46]
```diff
@@ -0,0 +1,94 @@
...
+    const startIcon = startIconProp || "";
+    const children = childrenProp || "";
+    const endIcon = endIconProp || "";
```
Comment (ghost) [inline src/components/common/Button/index.tsx]:
```suggestion
    const startIcon = startIconProp &&V <span> starticon <span/>;

    const endIcon = endIconProp || "";
```

**Pull request merged**